packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -37,10 +37,51 @@ import type {
 } from "./detachedFieldIndexTypes.js";
 import { makeDetachedFieldIndexCodec } from "./detachedFieldIndexCodecs.js";
 
+/**
+ * Readonly interface for {@link DetachedFieldIndex}.
+ */
+export interface ReadOnlyDetachedFieldIndex {
+	/**
+	 * Creates a deep clone of this DetachedFieldIndex.
+	 */
+	clone(): DetachedFieldIndex;
+
+	/**
+	 * Returns a field key for the given ID.
+	 * This does not save the field key on the index. To do so, call {@link createEntry}.
+	 */
+	toFieldKey(id: ForestRootId): FieldKey;
+
+	/**
+	 * Returns the FieldKey associated with the given id.
+	 * Returns undefined if no such id is known to the index.
+	 */
+	tryGetEntry(id: Delta.DetachedNodeId): ForestRootId | undefined;
+
+	/**
+	 * Returns the FieldKey associated with the given id.
+	 * Fails if no such id is known to the index.
+	 */
+	getEntry(id: Delta.DetachedNodeId): ForestRootId;
+
+	/**
+	 * Creates a snapshot of the current state of the DetachedFieldIndex.
+	 */
+	createSnapshot(): DetachedFieldIndexSnapshot;
+}
+
+export interface DetachedFieldIndexSnapshot {
```

Comment:
I find this type (and the associated API that creates it) a little bit confusing, since usually we use the word snapshot to mean "the actual snapshotted data" but this is not that, it's actually a (tiny) tool that lets you revert some index (that it is implicitly bound to) to some state.

What do you think of renaming `createSnapshot` to `checkpoint` and have it return a simple `() => void` directly, then just document what it does in the @return doc?